tooling/backend_interface/src/proof_system.rs

```diff
@@ -36,6 +36,18 @@ impl Backend {
         InfoCommand { crs_path: self.crs_directory() }.run(binary_path)
     }
 
+    /// If we cannot get a valid backend, returns the default backend which supports all the opcodes
+    /// and uses Plonk with width 3
+    /// The function also prints a message saying we could not find a backend
+    pub fn get_backend_info_or_default(&self) -> (Language, Option<BackendOpcodeSupport>) {
+        if let Ok(backend_info) = self.get_backend_info() {
+            (backend_info.0, Some(backend_info.1))
+        } else {
+            println!("No valid backend found, defaulting to Plonk with width 3 and all opcodes supported");
+            (Language::PLONKCSat { width: 3 }, None)
```

Comment:
I don't see how you want to avoid having a default language?

Reply:
I'm not suggesting that we shouldn't have a default language but that the default opcode support should be specified in a `BackendOpcodeSupport` struct rather than returning `None` and making it the responsibility of the caller to construct their own closure. This would allow us to remove a lot of the `if-else`s in nargo_cli.

Reply:
so this function should return `(Language, BackendOpcodeSupport)` rather than `(Language, Option<BackendOpcodeSupport>)`